ironfish/src/rpc/adapters/socketAdapter/socketAdapter.ts

```diff
@@ -193,6 +200,22 @@ export abstract class RpcSocketAdapter implements IRpcAdapter {
 
       const message = result.result.data
 
+      // Authentication
+      const requestAuthToken = message.auth
+      const rpcAuthTokenExists = await this.fileSystem.exists(this.rpcAuthTokenPath)
+
+      if (!rpcAuthTokenExists) {
+        this.logger.debug(`Missing RPC Auth token files at ${this.rpcAuthTokenPath}.`)
+        this.emitResponse(client, this.constructUnauthenticatedRequest())
+        return
+      }
+
+      const rpcAuthToken = await this.fileSystem.readFile(this.rpcAuthTokenPath)
+      if (requestAuthToken !== rpcAuthToken) {
+        this.emitResponse(client, this.constructUnauthenticatedRequest())
+        return
+      }
```

Comment:
If we move this into RpcServer, we wouldn't have to duplicate this for each adapter we want to do this.